pws/ddg.py

```diff
@@ -0,0 +1,108 @@
+#coding=utf-8
+
+from bs4 import BeautifulSoup
+from time import sleep as wait
+import re
+import requests
+try:
+    from html.parser import HTMLParser
+except ImportError:
+    from HTMLParser import HTMLParser
+
+
+##################################################
+# Copied code
+##################################################
+class MLStripper(HTMLParser):
+    # Code copied from StackOverflow http://stackoverflow.com/a/925630/3664835
+    def __init__(self):
+        self.reset()
+        self.strict = False
+        self.convert_charrefs= True
+        self.fed = []
+    def handle_data(self, d):
+        self.fed.append(d)
+    def get_data(self):
+        return ''.join(self.fed)
+
+
+def strip_tags(html):
+    # Code copied from StackOverflow http://stackoverflow.com/a/925630/3664835
+    s = MLStripper()
+    s.feed(html)
+    return ' '.join(s.get_data().split())
+
+
+##################################################
+# Helpers
+##################################################
+# https://duckduckgo.com/html/?q=hello%20world
+def generate_url(query):  # , num, start, recent, country_code):
+    """(str, str, str) -> str
+    A url in the required format is generated.
+    """
+    query = '%20'.join(query.split())
+    url = 'https://duckduckgo.com/html/?q=' + query
+    return url
+
+
+##################################################
+# Class
+##################################################
+class Ddg:
+
+    @staticmethod
+    def search(query, num=10, start=0, sleep=True, recent=None, country_code=None):
+        if sleep:
+            wait(1)
+        url = generate_url(query)
```

Comment:
I used query: line 58
num & start: line 60
sleep: line 56

duckduckgo does not have a news search feature and does not give a date to
it's search results, so I did not use recent.

 country_code is not set via the url.  I suppose there is a way to
initialize the country by creating a session, but this is above my skill
set.

I just needed the code to do some work for me so I thought I would pay back
the favor.

On Sat, Mar 5, 2016 at 10:18 PM, Rohith PR <notifications@github.com> wrote:

> In pws/ddg.py
> <https://github.com/rohithpr/py-web-search/pull/22#discussion_r55132459>:
>
> > +    """
> > +    query = '%20'.join(query.split())
> > +    url = 'https://duckduckgo.com/html/?q=' + query
> > +    return url
> > +
> > +
> > +##################################################
> > +# Class
> > +##################################################
> > +class Ddg:
> > +
> > +    @staticmethod
> > +    def search(query, num=10, start=0, sleep=True, recent=None, country_code=None):
> > +        if sleep:
> > +            wait(1)
> > +        url = generate_url(query)
>
> You're using only query. What about other arguments?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rohithpr/py-web-search/pull/22/files#r55132459>.
>